rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -24,6 +24,19 @@
 #include "rosbag2_cpp/logging.hpp"
 #include "rosbag2_cpp/readers/sequential_reader.hpp"
 
+namespace
+{
+void fill_topics_and_types(
+  const rosbag2_storage::BagMetadata & metadata,
+  std::vector<rosbag2_storage::TopicMetadata> & topics_and_types)
+{
+  topics_and_types.reserve(metadata.topics_with_message_count.size());
```

Comment:
should we warn here if the incoming vector is not empty?

Reply:
I think we should clear it. Updated the PR to do that, lmk if you disagree with that approach.

Reply:
looks good.